crates/sui-core/src/consensus_adapter.rs

```diff
@@ -1287,6 +1301,55 @@ impl SubmitToConsensus for Arc<ConsensusAdapter> {
         self.submit_batch(transactions, None, epoch_store)
             .map(|_| ())
     }
+
+    fn submit_best_effort(
+        &self,
+        transaction: &ConsensusTransaction,
+        epoch_store: &Arc<AuthorityPerEpochStore>,
+        // timeout is required, or the spawned task can run forever
+        timeout: Duration,
```

Comment:
I suggest not use a timeout here. The logic wrapped in a timeout is all local logic, and I'm not absolutely sure if they are cancellation safe.